android/sdl_android/src/main/java/com/smartdevicelink/transport/TCPTransportManager.java

```diff
@@ -0,0 +1,136 @@
+package com.smartdevicelink.transport;
+
+import android.os.Handler;
+import android.os.Message;
+import android.util.Log;
+
+import com.smartdevicelink.protocol.SdlPacket;
+import com.smartdevicelink.protocol.SdlProtocol;
+import com.smartdevicelink.transport.enums.TransportType;
+import com.smartdevicelink.transport.utl.TransportRecord;
+import com.smartdevicelink.util.DebugTool;
+
+import java.lang.ref.WeakReference;
+
+public class TCPTransportManager extends TransportManagerBase{
+
+    private static final String TAG = "TCPTransportManager";
+
+    TCPHandler tcpHandler;
+    private MultiplexTcpTransport transport;
+    private TCPTransportConfig config;
+
+    public TCPTransportManager(TCPTransportConfig config, TransportEventListener transportEventListener){
+        super(config,transportEventListener);
+        Log.d(TAG, "USING THE TCP TRANSPORT MANAGER");
+        this.config = config;
+        tcpHandler = new TCPHandler(this);
+        transport = new MultiplexTcpTransport(config.getPort(), config.getIPAddress(),config.getAutoReconnect(),tcpHandler, null);
+    }
+
+    @Override
+    public void start() {
+        transport.start();
+
+    }
+
+    @Override
+    public void close(long sessionId) {
+        transport.stop();
+    }
+
+    @Override
+    public void resetSession() {
+        if(transport != null){
+            transport.stop();
+        }
+        //TODO make sure this makes sense
+        transport = new MultiplexTcpTransport(config.getPort(), config.getIPAddress(),config.getAutoReconnect(),null, null);
```

Comment:
Should the handler be `null` here? in `MultiplexBaseTransport` the `setState` method calls handler without a `null` check, it could be a potential NPE.